inc/classes/admin/settings/class-page.php

```diff
@@ -0,0 +1,1045 @@
+<?php
+namespace WP_Rocket\Admin\Settings;
+
+defined( 'ABSPATH' ) || die( 'Cheatin&#8217; uh?' );
+
+/**
+ * Registers the admin page and WP Rocket settings
+ *
+ * @since 3.0
+ * @author Remy Perona
+ */
+class Page {
+	/**
+	 * Plugin slug
+	 *
+	 * @since 3.0
+	 * @author Remy Perona
+	 *
+	 * @var string
+	 */
+	private $slug;
+
+	/**
+	 * Plugin page title
+	 *
+	 * @since 3.0
+	 * @author Remy Perona
+	 *
+	 * @var string
+	 */
+	private $title;
+
+	/**
+	 * Required capability to access the page
+	 *
+	 * @since 3.0
+	 * @author Remy Perona
+	 *
+	 * @var string
+	 */
+	private $capability;
+
+	/**
+	 * Settings instance
+	 *
+	 * @since 3.0
+	 * @author Remy Perona
+	 *
+	 * @var Settings
+	 */
+	private $settings;
+
+	/**
+	 * Render implementation
+	 *
+	 * @since 3.0
+	 * @author Remy Perona
+	 *
+	 * @var \WP_Rocket\Interfaces\Render_Interface
+	 */
+	private $render;
+
+	/**
+	 * Constructor
+	 *
+	 * @since 3.0
+	 * @author Remy Perona
+	 *
+	 * @param array                                  $args     Array of required arguments to add the admin page.
+	 * @param Settings                               $settings Instance of Settings class.
+	 * @param \WP_Rocket\Interfaces\Render_Interface $render   Implementation of Render interface.
+	 */
+	public function __construct( $args, Settings $settings, \WP_Rocket\Interfaces\Render_Interface $render ) {
+		$this->slug       = $args['slug'];
+		$this->title      = $args['title'];
+		$this->capability = $args['capability'];
+		$this->settings   = $settings;
+		$this->render     = $render;
+	}
+
+	/**
+	 * Registers the class and hooks
+	 *
+	 * @since 3.0
+	 * @author Remy Perona
+	 *
+	 * @param array                                  $args     Array of required arguments to add the admin page.
+	 * @param Settings                               $settings Instance of Settings class.
+	 * @param \WP_Rocket\Interfaces\Render_Interface $render   Implementation of Render interface.
+	 * @return void
+	 */
+	public static function register( $args, Settings $settings, $render ) {
+		$self = new self( $args, $settings, $render );
+
+		add_action( 'admin_menu', [ $self, 'add_admin_page' ] );
+		add_action( 'admin_init', [ $self, 'configure' ] );
+		add_filter( 'option_page_capability_' . $self->slug, [ $self, 'required_capability' ] );
```

Comment:
It will be hell for someone who want to remove these hooks :/

Reply:
In this case, do we really want to give the possibility to unhook?

Reply:
You never know why people want to unhook things, so I tend to say yes.
On the other hand, since it doesn't seem easy to expose `$self`, I'd say don't touch it until the day it is needed.